### DISCUSSION TITLE
Extremely high IO/sysload post-index

### DISCUSSION THREAD

**Discussion by JayDub72** [photoprism/photoprism#2628] [Q&A]
I am new to Photoprism but trying it out for our family photo management solution.  I have been running it for quite a few days and have noticed a pattern of extremely high IO, sysload, and network traffic happening on the box.  I enabled the logs at debug level and the only thing I see happening is facial recognition updates. The persistent consumption of resources seems excessive.  I am sure that it is part of my configuration but I was hoping that someone may shed some light on a way for me to change a configuration to help alleviate.  

I recognize that I could disable facial recognition but that is one of the key features that I am looking for in the photo management solution.

Here are things that I think would be helpful but please feel free to ask any questions about my configuration that would help.

Running on VM running Ubuntu 22.04.01 LTS on Proxmox 7.2-7.

VM has been allocated 8 CPUs and 8GB of RAM. The mount points for the pictures are on an NFS share (which is where I am guessing the issue comes in).   The database is on a local filesystem and not the NFS mount, however.

Thanks in advance for your help!

Jason

The docker compose YML file is:

services:
  photoprism:
    image: photoprism/photoprism:latest
    depends_on:
      - mariadb
    security_opt:
      - seccomp:unconfined
      - apparmor:unconfined
    ports:
      - "2342:2342" # HTTP port (host:container)
    environment:
      PHOTOPRISM_ADMIN_PASSWORD: "SECRET"          # !!! PLEASE CHANGE YOUR INITIAL "admin" PASSWORD !!!
      PHOTOPRISM_SITE_URL: "http://localhost:2342/"  # public server URL incl http:// or https:// and /path, :port is optional
      PHOTOPRISM_ORIGINALS_LIMIT: 5000               # file size limit for originals in MB (increase for high-res video)
      PHOTOPRISM_HTTP_COMPRESSION: "gzip"            # improves transfer speed and bandwidth utilization (none or gzip)
      PHOTOPRISM_DEBUG: "false"                      # run in debug mode (shows additional log messages)
      PHOTOPRISM_PUBLIC: "false"                     # no authentication required (disables password protection)
      PHOTOPRISM_READONLY: "false"                   # don't modify originals directory (reduced functionality)
      PHOTOPRISM_EXPERIMENTAL: "false"               # enables experimental features
      PHOTOPRISM_DISABLE_CHOWN: "false"              # disables storage permission updates on startup
      PHOTOPRISM_DISABLE_WEBDAV: "false"             # disables built-in WebDAV server
      PHOTOPRISM_DISABLE_SETTINGS: "false"           # disables Settings in Web UI
      PHOTOPRISM_DISABLE_TENSORFLOW: "false"         # disables all features depending on TensorFlow
      PHOTOPRISM_DISABLE_FACES: "false"              # disables facial recognition
      PHOTOPRISM_DISABLE_CLASSIFICATION: "false"     # disables image classification
      PHOTOPRISM_DARKTABLE_PRESETS: "false"          # enables Darktable presets and disables concurrent RAW conversion
      PHOTOPRISM_DETECT_NSFW: "false"                # flag photos as private that MAY be offensive (requires TensorFlow)
      PHOTOPRISM_UPLOAD_NSFW: "true"                 # allows uploads that MAY be offensive
      # PHOTOPRISM_DATABASE_DRIVER: "sqlite"         # SQLite is an embedded database that doesn't require a server
      PHOTOPRISM_DATABASE_DRIVER: "mysql"            # use MariaDB 10.5+ or MySQL 8+ instead of SQLite for improved performance
      PHOTOPRISM_DATABASE_SERVER: "mariadb:3306"     # MariaDB or MySQL database server (hostname:port)
      PHOTOPRISM_DATABASE_NAME: "photoprism"         # MariaDB or MySQL database schema name
      PHOTOPRISM_DATABASE_USER: "photoprism"         # MariaDB or MySQL database user name
      PHOTOPRISM_DATABASE_PASSWORD: "SECRET"       # MariaDB or MySQL database user password
      PHOTOPRISM_SITE_TITLE: "PhotoPrism"
      PHOTOPRISM_SITE_CAPTION: "AI-Powered Photos App"
      PHOTOPRISM_SITE_DESCRIPTION: ""
      PHOTOPRISM_SITE_AUTHOR: ""
      HOME: "/photoprism"
    working_dir: "/photoprism"
    volumes:
      - "/mnt/media/photos/storage:/photoprism/storage"
      - "/mnt/media/photos/imports:/photoprism/imports"
      - "/mnt/media/photos/originals:/photoprism/originals"

  mariadb:
    restart: unless-stopped
    image: mariadb:10.6
    security_opt:
      - seccomp:unconfined
      - apparmor:unconfined
    command: mysqld --innodb-buffer-pool-size=128M --transaction-isolation=READ-COMMITTED --character-set-server=utf8mb4 --collation-server=utf8mb4_unicode_ci --max-conne
ctions=512 --innodb-rollback-on-timeout=OFF --innodb-lock-wait-timeout=120
    volumes:
      - "./database:/var/lib/mysql" # important, don't remove
    environment:
      MYSQL_ROOT_PASSWORD: SECRET
      MYSQL_DATABASE: photoprism
      MYSQL_USER: photoprism
      MYSQL_PASSWORD: SECRET

**Comment by JayDub72** [comment]
BTW -- attached is a screenshot of the performance metrics from Grafana:

![Screen Shot 2022-08-21 at 1 52 24 PM](https://user-images.githubusercontent.com/31083720/185806550-316643ba-8363-45e0-8884-becf4ca88557.png)

![Screen Shot 2022-08-21 at 1 54 34 PM](https://user-images.githubusercontent.com/31083720/185806576-7794abeb-3931-46e0-93e1-3920976584c5.png)
![Screen Shot 2022-08-21 at 1 55 05 PM](https://user-images.githubusercontent.com/31083720/185806591-965e8fea-f676-427c-a003-59d9096be994.png)

**Comment by graciousgrey** [comment]
It is expected that the background worker is running frequently.

Maybe our performace troubleshooting guide can help you reduce the system load: https://docs.photoprism.app/getting-started/troubleshooting/performance/#troubleshooting
